web-gateway/app/controllers/ItemForm.java

```diff
@@ -29,10 +29,8 @@
 
     public List<ValidationError> validate() {
         List<ValidationError> errors = new ArrayList<>();
-        
+        Currency c = Currency.valueOf(currency);
         try {
-            Currency c = Currency.valueOf(currency);
-
             if (!c.isValidStep(increment.doubleValue())) {
```

Comment:
I think the `Currency c = Currency.valueOf(currency);` statement should be inside the `try{} catch{}` block because providing the currency `ASDF` will cause to throw an `IllegalArgumentException`.

That will require your other change (below) to be moved inside this `try{} catch` .